File: squares-of-a-sorted-array/squares-of-a-sorted-array.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        var result = new int[nums.Length];
        for (var i = 0; i < nums.Length; i++)
        {
            var abs = Math.Abs(nums[i]);

            int pos = 0;

            for (var j = 0; j < nums.Length; j++)
            {
                if (abs > Math.Abs(nums[j]))
                {
                    pos++;
                }
            }
            while (result[pos] != 0)
            {
                pos++;
            }
            result[pos] = abs * abs;
        }


        return result;
    }
}

File: squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int> &nums)
    {
        vector<int> result(nums.size());
        int l = 0, r = nums.size() - 1, ptr = nums.size() - 1;

        while (ptr >= 0)
        {
            result[ptr--] = abs(nums[l]) > abs(nums[r]) ? nums[l] * nums[l++] : nums[r] * nums[r--];
        }

        return result;
    }
};